internal/tool/generate/generator_test.go

```diff
@@ -237,6 +238,80 @@ func TestGenerator(t *testing.T) {
 	}
 }
 
+// TestGeneratorBuildWithTags runs "weaver generate" on all the files in
+// testdata/tags and checks if the command succeeds. Each file should have some build tags.
+func TestGeneratorBuildWithTags(t *testing.T) {
```

Comment:
Why can't we use runGenerator for this test?

Reply:
runGenerator copies every single file in a temp directory, and tries to build it. I chose this approach because I wanted to run go build on all the files in the directory and check that weaver_gen contains things only for the good files. That adds a little bit of boilerplate code for sure. If you have a strong preference, I can switch to calling runGenerator instead.

Reply:
I am fine with you keeping what you have, but I don't quite understand the downside of using runGenerator. It would copy all of the files and build it. You can still check that weaver_gen.go does not contain the components defined in the bad file.

Reply:
Done